isisd/isisd.c

```diff
@@ -3249,13 +3249,73 @@ void isis_area_overload_on_startup_set(struct isis_area *area,
 }
 
 void isis_area_advertise_high_metrics_set(struct isis_area *area,
-				       bool advertise_high_metrics)
+					  bool advertise_high_metrics, int restart_metric)
 {
 	if (area->advertise_high_metrics != advertise_high_metrics) {
-		area->advertise_high_metrics = advertise_high_metrics;
+		if (advertise_high_metrics) {
+			struct listnode *node;
+			struct isis_circuit *circuit;
+			int max_metric;
+
+			if (!restart_metric) {
+				if (area->oldmetric && area->newmetric) {
+					max_metric = ISIS_NARROW_METRIC_INFINITY;
+				} else if (area->newmetric) {
+					max_metric = MAX_WIDE_LINK_METRIC;
+				} else {
+					max_metric = MAX_NARROW_LINK_METRIC;
+				}	
```

Comment:
If else is sufficient
```
				if (!area->oldmetric && area->newmetric)
					max_metric = MAX_WIDE_LINK_METRIC;
				else
					max_metric = MAX_NARROW_LINK_METRIC;
```
No bracket when only one line

Reply:
Done